### ISSUE TITLE
Problem with spaces in SKU and GetMyFeesEstimateForSKU

### ISSUE THREAD

**Issue by GuybrushX** [abuzuhri/Amazon-SP-API-CSharp#35]
I have the following SKU (it is a set of 2 different articles):
"SKU1 + SKU2-FBA" (without the quotation marks)

I get the following error - I think because of some URL encoding issues -> simple SKUs like "SKU1-FBA" are working fine:
```
message: "The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.

The Canonical String for this request should have been
'POST
/products/fees/v0/listings/SKU1%2520%2B%2520SKU2-FBA/feesEstimate

host:sellingpartnerapi-eu.amazon.com
x-amz-access-token:Token
x-amz-date:Timestamp
x-amz-security-token:Token

host;x-amz-access-token;x-amz-date;x-amz-security-token
Some Hex Value'

The String-to-Sign should have been
'AWS4-HMAC-SHA256
Timestamp
20211218/eu-west-1/execute-api/aws4_request
Some Hex Value'

**Comment by abuzuhri** [comment]
Hi @GuybrushX 

I confirm the issue and i will solve it very soon

**Comment by abuzuhri** [comment]
Hi @GuybrushX 

After checking i found your URL have double encoding (/products/fees/v0/listings/SKU1%2520%2B%2520SKU2-FBA/feesEstimate)

to solve the problem just send the SKU without encoding and library will do encoding automatically 

I test the problem and its work in my side

**Comment by abuzuhri** [comment]
> Hi @GuybrushX
> 
> After checking i found your URL have double encoding (/products/fees/v0/listings/SKU1%2520%2B%2520SKU2-FBA/feesEstimate)
> 
> to solve the problem just send the SKU without encoding and library will do encoding automatically
> 
> I test the problem and its work in my side

@GuybrushX  do you have time to recheck this

**Comment by GuybrushX** [comment]
Hi. Thank you for your fast reply and investigation.

Did you create and test with a SKU like I described?

I have no idea where this double encoding is coming from since the message I pasted is straight from Visual Studio -> I didn't use any manual encoding.

If I understand that message correctly it is even asking for double encoding since it says:
The Canonical String for this request should have been
'POST
/products/fees/v0/listings/SKU1%2520%2B%2520SKU2-FBA/feesEstimate
...'

The "ExecuteRequest" method seems to use the following path for RestClient.Execute<T>(Request):
/products/fees/v0/listings/SKU1 + SKU2-FBA/feesEstimate

So there isn't any encoding applied yet.

Any idea how to fix it?

**Comment by abuzuhri** [comment]
Hello @GuybrushX ,

> Did you create and test with a SKU like I described?
I create Sku (88 554455544) so URL like that  (/products/fees/v0/listings/88 554455544/feesEstimate) , and its work with me,  

Use this website https://www.urldecoder.org/ to decode The Url (/products/fees/v0/listings/SKU1%2520%2B%2520SKU2-FBA/feesEstimate) you will see that you must do decoding 2 times to get (/products/fees/v0/listings/SKU1 + SKU2-FBA/feesEstimate)


I will try to create same SKU (SKU1 + SKU2-FBA) and reply here again

**Comment by abuzuhri** [comment]
> Did you create and test with a SKU like I described?

Now i do same SKU and i got same error .
but its almost this bug in Amazon SP API , i will double check and its bug in amazon SP API i will open bug there

**Comment by abuzuhri** [comment]
Bug Open in [Amazon SP API](https://github.com/amzn/selling-partner-api-docs/issues/1832)

**Comment by GuybrushX** [comment]
> Bug Open in [Amazon SP API](https://github.com/amzn/selling-partner-api-docs/issues/1832)

Thank you for double-checking and confirming. I hope Amazon will reply faster than usual with this very specific case :-)

**Comment by J-W-Chan** [comment]
i fixed ths bug  on #85

**Comment by abuzuhri** [comment]
I close this bug after fix from @J-W-Chan  thanks
